### DISCUSSION TITLE
Could I run...

### DISCUSSION THREAD

**Discussion by yuriv** [teal-language/tl#482] [Q&A]
I have script tl_using.lua:

` local tl = require("tl") 
  tl.loader()
 
  local function property() : number
    return 34.3
  end
  
  value = property()
  print(value)`

I try to run it like this
`lua tl_using.lua 
lua: tl_using.lua:6: unexpected symbol near ':'`

I couldn't run tl code with lua interpreter?

**Comment by lenscas** [comment]
after doing
```lua
local tl = require"tl"
tl.loader()
```
the lua vm becomes able to load `.tl` files. 

However, what you seem to have done is put the teal code inside the same file as the lua code that loads teal. This does not work, as lua first needs to parse the file before it can run. Because your file contains teal syntax that is invalid lua, the lua vm can not run the code at all and thus never even gets to the part where it loads the teal compiler.

**Reply by yuriv** [reply]
Thank you for reply. But what I need to do? How I can run this simple script? I need to move tl code to the separate .tl file. And how I can load it from .lua file?

**Reply by lenscas** [reply]
you can either run a .tl file directly by running `tl run tl_using.tl`
or you can use lua by making a file like
```lua
local tl = require"tl"
tl.loader()
require("tl_using")
```
and put the teal code you want to run in the `tl_using` file

**Reply by yuriv** [reply]
This is my code:

-- prop.tl
local function property() : number
return 34.3
end

return { property = property }

-- tl_using.lua
local tl = require("tl") 
tl.loader()

require "prop"
print(property())

After run a have got:
lua: ./tl.lua:9443: bad argument #1 to 'load' (function expected, got string)
stack traceback:
	[C]: in function 'load'
	./tl.lua:9443: in function <./tl.lua:9417>
	[C]: in function 'require'
	tl_using.lua:5: in main chunk
	[C]: ?

Lua 5.1.4  Copyright (C) 1994-2008 Lua.org, PUC-Rio

What the problem?
